### PR TITLE
PLACEHOLDER: Support services/classes design.

### PR DESCRIPTION
**Pull request by clstokes**
## Do not merge!

This PR is meant to highlight the suggested design/packaging for implementing functionality for Provider Support services such as opening, closing, and replying to support tickets.

### CODE REVIEW

File: src/main/java/org/dasein/cloud/resource/ResourceServices.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.dasein.cloud.resource;
+
+import javax.annotation.Nullable;
+
+/**
+ * User: Eugene Yaroslavtsev
+ * Date: 20.08.2014
+ */
+public interface ResourceServices {
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/resource/ResourceServices.java]: I think we should remove this interface/implementation and move `SupportService` to [PlatformServices](https://github.com/TheWeatherCompany/dasein-cloud-core/blob/92a8fc01c09e3665556d323d49f0c4bf885bafaa/src/main/java/org/dasein/cloud/platform/PlatformServices.java).

File: src/main/java/org/dasein/cloud/resource/SupportService.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import org.dasein.cloud.resource.model.options.*;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+/**
+ * User: Eugene Yaroslavtsev
+ * Date: 20.08.2014
+ */
+public interface SupportService {
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/resource/SupportService.java]: Given other uses of the term _Support_, I think we should rename this to TicketSupport.

File: src/main/java/org/dasein/cloud/resource/SupportService.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * Gets the list of replies in tickets having the specified options
+     *
+     * @param options the specified options
+     * @return the replies with cc emails
+     * @throws InternalException
+     * @throws CloudException
+     */
+    public ReplyWithCcEmails listReplies(@Nonnull TicketListRepliesOptions options) throws InternalException, CloudException;
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/resource/SupportService.java]: Does `ReplyWithCcEmails` make sense here or `Collection<Reply>` instead?

Comment (eyaroslavtsev) [inline src/main/java/org/dasein/cloud/resource/SupportService.java]: I think you're right. This method should return `Collection<Reply>`

File: src/main/java/org/dasein/cloud/resource/SupportService.java (hunk 2 above)
Comment (clstokes) [inline src/main/java/org/dasein/cloud/resource/SupportService.java]: We should prefix all the base model classes (AttachmentData, Attachment, etc) with _Ticket\_*.

File: src/main/java/org/dasein/cloud/resource/model/options/TicketListRepliesOptions.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * User: Eugene Yaroslavtsev
+ * Date: 21.08.2014
+ */
+public class TicketListRepliesOptions {
+
+    private String afterTime;
+    private String beforeTime;
+    private String ticketId;
+    private String maxResults;
+    private String nextToken;
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/resource/model/options/TicketListRepliesOptions.java]: Is `maxResults` and `nextToken` consistent with other services that support paging?

Comment (eyaroslavtsev) [inline src/main/java/org/dasein/cloud/resource/model/options/TicketListRepliesOptions.java]: We can remove `maxResults` and `nextToken` from our *options classes. i think this parameters do not need us in `core`

File: src/main/java/org/dasein/cloud/platform/support/TicketService.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * Gets the list of tickets having the specified options
+     *
+     * @param options the specified options
+     * @return the list of tickets
+     * @throws InternalException
+     * @throws CloudException
+     */
+    public Collection<Ticket> listTickets(@Nonnull TicketListOptions options) throws InternalException, CloudException;
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/platform/support/TicketService.java]: Change all `Collection...` return values to `Iterable` per dasein project standards.

**Comment by clstokes** [comment]
1. See comments in https://github.com/greese/dasein-cloud-core/pull/270 for other changes dasein project will expect.

**Pull request closed without merging**